apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -445,116 +465,221 @@ func setOverrideSpec(override **route.OverrideSpec, anno map[string]string) {
 // DefaultServices - common function for calling individual services' defaulting functions
 func (r *OpenStackControlPlane) DefaultServices() {
 	// Cinder
-	r.Spec.Cinder.Template.Default()
-	initializeOverrideSpec(&r.Spec.Cinder.APIOverride.Route, true)
-	r.Spec.Cinder.Template.SetDefaultRouteAnnotations(r.Spec.Cinder.APIOverride.Route.Annotations)
+	if r.Spec.Cinder.Enabled || r.Spec.Cinder.Template != nil {
+		if r.Spec.Cinder.Template == nil {
+			r.Spec.Cinder.Template = &cinderv1.CinderSpecCore{}
+		}
+		r.Spec.Cinder.Template.Default()
+		initializeOverrideSpec(&r.Spec.Cinder.APIOverride.Route, true)
+		r.Spec.Cinder.Template.SetDefaultRouteAnnotations(r.Spec.Cinder.APIOverride.Route.Annotations)
+	}
 
 	// Galera
-	for key, template := range r.Spec.Galera.Templates {
-		if template.StorageClass == "" {
-			template.StorageClass = r.Spec.StorageClass
+	if r.Spec.Galera.Enabled && r.Spec.Galera.Templates != nil {
+		if r.Spec.Galera.Templates == nil {
```

Comment:
This seems wrong. It cannot be that both the first and the second' "if" is true at the same time as that would require that the template is nil and not nil at the same time.

Reply:
right thats wrong